### ISSUE TITLE
usage with starlette

### ISSUE THREAD

**Issue by HedgeShot** [chunky-monkeys/keycloak-client#45]
Hi,

How can I use this package with Starlette (FastApi)?
I tried as shown in the docs but it says `keycloak.json` is not found.
Where do I need to put such a file & what do i need to put inside?

Thanks!

**Comment by HedgeShot** [comment]
Found out some of the settings to add to keycloak.json for those interested:

```
{
    "realm": "fastapi",
    "auth_server_url": "http://keycloak_url/auth",
    "ssl_required": "false",
    "resource": "fastapi",
    "verify_token_audience": "false",
    "credentials": {
        "secret": "client_secret_from_keycloak"
    },
    "confidential_port": 8080,
    "policy_enforcer": {}
}
```

Now my problem is that the route `/kc/callback` doesnt exist. My API has uri like `/api/v1/` prepended to all routes. Should i create the route myself?

**Comment by HedgeShot** [comment]
I could eventually make it work by changing the code in /extensions/starlette.py as `/kc/callback` is hardcoded. I suggest to be able to pass a new arg to `AuthenticationMiddleware` like `pathprefix` to allow for different routes.

**Comment by akhilputhiry** [comment]
> I could eventually make it work by changing the code in /extensions/starlette.py as `/kc/callback` is hardcoded. I suggest to be able to pass a new arg to `AuthenticationMiddleware` like `pathprefix` to allow for different routes.

you can pass callback url to the middleware.
Please see the following snippets.

* [Function definition](https://github.com/chunky-monkeys/keycloak-client/blob/master/src/keycloak/extensions/starlette.py#L49)
* [Usage](https://github.com/chunky-monkeys/keycloak-client/blob/master/tests/extensions/starlette.py#L15)

**Comment by HedgeShot** [comment]
yes you can but this line is in `starlette.py`:

```
if request.url.path == "/kc/callback":
```

so `request.url.path` will never match and `Callback()` never be called.

**Comment by akhilputhiry** [comment]
HI @HedgeShot 

So that line should be something like

```
 if request.url.path == self.callback_uri:
```

why dont you raise a PR with the change?

**Comment by HedgeShot** [comment]
never done any but will try.

**Comment by akhilputhiry** [comment]
This has been fixed in `2.0.2`
